### PR TITLE
🗿Circle 2.1

### PR DESCRIPTION
**Pull request by rye**
# What?

This PR updates our CircleCI configuration to Circle 2.1.

To-Do list:
- [x] Change out "commands" for actual commands, and use those in the job configurations.
- [x] DRY any commands up so that we don't have to look at multiple copies of the same thing.
- [x] Look at possibly extracting out into orbs so this CI config can be less "special" and more reusable.
- [x] DRY up workflows.

# Why?

The old `<<`- and `*`-based DRY config strategy is (i) hard to understand, and (ii) no longer the best possible option.  Circle added Commands, Executors, and such, so that we can re-use bits of our configuration.  Plus, 2.1 is greater than 2.0, so we're cooler if we change the numbers to be bigger. 😎

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,50 +1,79 @@
...
+  danger:
+    description: 'Run Danger'
+    steps:
+      - run:
+          command: if ! [[ $CIRCLE_PR_NUMBER ]]; then yarn run danger --id $task; fi
+          when: always
+          environment:
+            DANGER_DISABLE_TRANSPILATION: 'true'
+
+  embed-env-vars:
```
Comment (hawkrives) [inline .circleci/config.yml]: this one could just vanish, tbh

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,50 +1,79 @@
...
+    description: 'Embed environment variables into the app'
+    steps:
+      - run: touch .env.js
+
+  restore-cache-bundler:
+    description: 'Restore the Bundler cache'
+    steps:
+      - restore_cache:
+          key: 'v2-ruby-dependencies-{{ arch }}-{{ checksum "Gemfile.lock" }}'
```
Comment (hawkrives) [inline .circleci/config.yml]: These giant strings should use YAML references, I think, so we don't wind up (say) changing `save` to v4 and forgetting `restore` at v3. We can just annotate this first string like `key: &bundler-cache-key 'v2-…'`, and then use it in the `save` command as `*bundler-cache-key`.

IMO anyway.

File: .circleci/config.yml (hunk 1 above)
Comment (rye) [reply .circleci/config.yml]: @hawkrives are you saying we do not need this anymore?

File: .circleci/config.yml (hunk 2 above)
Comment (rye) [reply .circleci/config.yml]: That's out of scope for this PR, but a good idea.

File: .circleci/config.yml (hunk 1 above)
Comment (drewvolz) [reply .circleci/config.yml]: I agree with @hawkrives. We're no longer relying on the `.env.js` file since api keys have been moved to `ccc-server`.

Comment (rye) [reply .circleci/config.yml]: Removed.

**Comment by rye** [comment]
I think this is now ready-for-review in a more official sense. Obviously more work can be done to get us farther down the modularization path, but this MWE makes me happier.

File: .circleci/config.yml [hunk 3]
```diff
@@ -340,7 +345,6 @@ jobs:
       task: JS-bundle-android
     steps:
       - checkout
-      - attach_workspace: *attach-workspace-node_modules
```
Comment (hawkrives) [inline .circleci/config.yml:320]: Pretty sure we still want this one?

Comment (rye) [reply .circleci/config.yml:320]: Oh yeah

Comment (hawkrives) [reply .circleci/config.yml:320]: (insert kool-aid guy)

**Review by hawkrives — APPROVED** [review]
Sure

**Review by drewvolz — APPROVED** [review]
👍

**Pull request merged**